packages/framework/aqueduct/src/data-object-factories/pureDataObjectFactory.ts

```diff
@@ -259,10 +260,46 @@ export class PureDataObjectFactory<
 		return this.createNonRootInstanceCore(runtime, [this.type], initialState);
 	}
 
+	/**
+	 * Creates a new instance of the object.
+	 * @param runtime - container runtime. It is the runtime that will be used to create the object. It will produce
+	 * the underlying infrastructure to get the data object to operate.
+	 * @param initialState - The initial state to provide to the created component.
+	 * @param packagePath - The path to the data store factory to use to create the data object.
+	 * @returns an array containing the object created by this factory and an IDataStore object that enables users to
+	 * alias the data object.
+	 * The data object is attached only when it is attached to the handle graph that connects to an aliased object or
+	 * when the data object is aliased.
+	 */
+	public async constructInstance(
+		containerRuntime: IContainerRuntimeBase,
+		initialState?: I["InitialState"],
+		packagePath?: Readonly<string[]>,
+	): Promise<[TObj, IDataStore]> {
```

Comment:
I don't think we should add this API is addition to the other APIs that already exist. I would keep one or the other. If we want to expose this pattern for creating objects here, I would deprecate the existing APIs and make sure that we move to this API.
Keeping both is just confusing and this API may end up never being used but we will have to keep it around forever.

Reply:
Basically, be clear on what pattern we want to expose, make sure there is a plan to transition to that and add / deprecate / remove APIs accordingly.

Reply:
My suggestion would be to separate this part of the change from the `createRootInstance` API because this will require more thinking and discussion.
@vladsud FYI

Reply:
I'm going with the direction of using this API and deprecating the `createRootInstance` api as it does not necessarily make it easier for the customer to decide its behavior. Talked with another customer, and they preferred the flexibility of the `createInstanceWithDataStore` api over the `createRootInstance` api. Also we would need to add packagePath to `createRootInstance`. That's not particularly complicated, but it may confuse other customers.